src/modules/emote_menu/components/Emotes.jsx

```diff
@@ -0,0 +1,303 @@
+        className={styles.emotesContainer}
+        onHeaderChange={handleHeaderChange}
+        ref={ref}
+      />
+    );
+  }
+);
+
+const SearchedEmotes = React.forwardRef(({search, onClick, cords, setCords, setRowColumnCounts, setSelected}, ref) => {
+  const emotes = useMemo(() => emoteStore.search(search), [search]);
+
+  const handleMouseOver = useCallback((newCords) => {
+    if (navigationMode === NavigationModeTypes.MOUSE) {
+      setCords(newCords);
+    }
+  }, []);
+
+  useEffect(() => {
+    const rowColumnCounts = [];
+    let foundFirstEmote = false;
+
+    for (const row of emotes) {
+      rowColumnCounts.push(!Array.isArray(row) ? 0 : row.length);
+
+      if (Array.isArray(row) && !foundFirstEmote) {
+        setSelected(row[0].item);
+        foundFirstEmote = true;
+      }
+    }
+    setRowColumnCounts(rowColumnCounts);
+  }, [emotes]);
+
+  const renderRow = useCallback(
+    ({key, style, index: y, className}) => {
+      const row = emotes[y];
+      if (row == null) return null;
+      return (
+        <div key={key} style={style} className={classNames(className, styles.row)}>
+          {row.map(({item}, x) => (
+            <Emote
+              emote={item}
+              onClick={onClick}
+              onMouseOver={() => handleMouseOver({x, y})}
+              active={x === cords.x && y === cords.y}
+            />
+          ))}
+        </div>
+      );
+    },
+    [emotes, cords, onClick]
+  );
+
+  if (emotes.length === 0) {
+    return (
+      <div className={styles.empty}>
+        {Icons.HEART_BROKEN}
+        <p>No results...</p>
+      </div>
+    );
+  }
+
+  return (
+    <VirtualizedList
+      ref={ref}
+      rowHeight={RowHeight}
+      windowHeight={WindowHeight}
+      totalRows={emotes.length}
+      renderRow={renderRow}
+      className={classNames(styles.emotesContainer, styles.searched)}
+      stickyRows={[]}
+    />
+  );
+});
+
+export default function renderEmotes(props) {
+  const {search, arrowKeys} = props;
+
+  const [cords, setCords] = useState({x: 0, y: 0});
+  const [rowColumnCounts, setRowColumnCounts] = useState([]);
+
+  const wrapperRef = useRef(null);
+
+  const isSearch = search.length > 0;
+
+  useEffect(() => {
+    function callback() {
+      navigationMode = NavigationModeTypes.MOUSE;
+    }
+
+    window.addEventListener('mousemove', callback);
+
+    return () => {
+      window.removeEventListener('mousemove', callback);
+    };
+  }, []);
+
+  useEffect(() => {
+    navigationMode = NavigationModeTypes.ARROW_KEYS;
+
+    let newCords = null;
+
+    switch (true) {
+      case arrowKeys.top:
+        newCords = travelUp(rowColumnCounts, cords);
+        break;
+      case arrowKeys.down:
+        newCords = travelDown(rowColumnCounts, cords);
+        break;
+      case arrowKeys.right:
+        newCords = travelRight(rowColumnCounts, cords);
+        break;
+      case arrowKeys.left:
+        newCords = travelLeft(rowColumnCounts, cords);
+        break;
```

Comment:
we don't seem to support other keys, like home or end. we can we support more of the keyboard nav buttons?